### ISSUE TITLE
omr promotion Test-sanity.functional x86 -Xgcpolicy:balanced crash vmState=0x0005ff06

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#4939]
Occurs in omr promotion build
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64/536/
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/571/
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-win_x86-64_cmprssptrs/516/
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-osx_x86-64_cmprssptrs/198

https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK8-linux_x86-64/674/
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK8-linux_x86-64_cmprssptrs/676/
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK8-win_x86-64_cmprssptrs/633/
https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK8-osx_x86-64_cmprssptrs/187/

```
15:19:16   [ERR] Type=Segmentation error vmState=0x0005ff06
15:19:16   [ERR] Windows_ExceptionCode=c0000005 J9Generic_Signal=00000004 ExceptionAddress=0000000068CB2232 ContextFlags=0010005f
15:19:16   [ERR] Handler1=00000000686F7AA0 Handler2=00007FFC2479ECF0 InaccessibleReadAddress=0000000000000000
15:19:16   [ERR] RDI=00007FF73AF79130 RSI=00007FF73AE54590 RAX=00000000000024FE RBX=00007FF73AF78050
15:19:16   [ERR] RCX=00007FF73AE52FA0 RDX=0000000000000000 R8=00007FF73AE52FA0 R9=0000000000000000
15:19:16   [ERR] R10=0000000000001000 R11=00007FF73AF78050 R12=0000000000000000 R13=0000000000000000
15:19:16   [ERR] R14=00007FF73AF78050 R15=00007FF73AE54590
15:19:16   [ERR] RIP=0000000068CB2232 RSP=00000048B6DEB610 RBP=00007FF73AE55F80 GS=002B
15:19:16   [ERR] FS=0053 ES=002B DS=002B
15:19:16   [ERR] XMM0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM2 43f0000000000000 (f: 0.000000, d: 1.844674e+19)
15:19:16   [ERR] XMM3 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM4 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
15:19:16   [ERR] Module=C:\Users\jenkins\workspace\Test-sanity.functional-JDK11-win_x86-64_cmprssptrs\openjdkbinary\j2sdk-image\bin\compressedrefs\j9jit29.dll
15:19:16   [ERR] Module_base_address=0000000068670000 Offset_in_DLL=0000000000642232
15:19:16   [ERR] 
15:19:16   [ERR] Method_being_compiled=com/ibm/jit/JITHelpers.getCharFromArrayByIndex(Ljava/lang/Object;I)C
15:19:16   [ERR] Target=2_90_20190228_612 (Windows Server 2012 R2 6.3 build 9600)
15:19:16   [ERR] CPU=amd64 (8 logical CPUs) (0x1ffb9c000 RAM)
15:19:16   [ERR] ----------- Stack Backtrace -----------
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x57ef72 (0x0000000068CB2232 [j9jit29+0x642232])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x5a390e (0x0000000068CD6BCE [j9jit29+0x666bce])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x59accf (0x0000000068CCDF8F [j9jit29+0x65df8f])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x570c84 (0x0000000068CA3F44 [j9jit29+0x633f44])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x5a6cd6 (0x0000000068CD9F96 [j9jit29+0x669f96])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x571142 (0x0000000068CA4402 [j9jit29+0x634402])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x596913 (0x0000000068CC9BD3 [j9jit29+0x659bd3])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x570c84 (0x0000000068CA3F44 [j9jit29+0x633f44])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x5703af (0x0000000068CA366F [j9jit29+0x63366f])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x2507d1 (0x0000000068983A91 [j9jit29+0x313a91])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x24f6cc (0x000000006898298C [j9jit29+0x31298c])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x245d4c (0x000000006897900C [j9jit29+0x30900c])
15:19:16   [ERR] Java_java_lang_invoke_MutableCallSite_invalidate+0x2659b2 (0x0000000068998C72 [j9jit29+0x328c72])
15:19:16   [ERR] (0x00000000686E56BC [j9jit29+0x756bc])
15:19:16   [ERR] (0x00000000686E8384 [j9jit29+0x78384])
15:19:16   [ERR] j9port_isCompatible+0x1ca26 (0x00007FFC2479F736 [j9prt29+0x1f736])
15:19:16   [ERR] j9port_isCompatible+0x1e124 (0x00007FFC247A0E34 [j9prt29+0x20e34])
15:19:16   [ERR] (0x00000000686E4E79 [j9jit29+0x74e79])
15:19:16   [ERR] (0x00000000686EA578 [j9jit29+0x7a578])
15:19:16   [ERR] (0x00000000686E9F58 [j9jit29+0x79f58])
15:19:16   [ERR] (0x00000000686F782F [j9jit29+0x8782f])
15:19:16   [ERR] j9port_isCompatible+0x1e15f (0x00007FFC247A0E6F [j9prt29+0x20e6f])
15:19:16   [ERR] (0x00000000686F744D [j9jit29+0x8744d])
15:19:16   [ERR] omrthread_get_category+0xa42 (0x00007FFC26AE4452 [J9THR29+0x4452])
15:19:16   [ERR] _o_strcat_s+0x5e (0x00007FFC24CDC1AE [ucrtbase+0x1c1ae])
15:19:16   [ERR] BaseThreadInitThunk+0x22 (0x00007FFC336A13D2 [KERNEL32+0x13d2])
15:19:16   [ERR] RtlUserThreadStart+0x34 (0x00007FFC35E954F4 [ntdll+0x154f4])
```

**Comment by andrewcraik** [comment]
@cathyzhyi since you are already looking at mutablecallsite for perf could you have a look at this? Note it si a blocker. @pshipton do we know what merged that caused the break?

**Comment by pshipton** [comment]
@andrewcraik these are the changes https://github.com/eclipse/omr/compare/b566b30...bd02dfd

**Comment by pshipton** [comment]
Sorry copied the wrong sha. This is the change https://github.com/eclipse/omr/compare/b566b30...fdc86e4

**Comment by pshipton** [comment]
@rwy0717 fyi

**Comment by 0xdaryl** [comment]
A fix for eclipse/omr#3604 was already merged this morning: eclipse/omr#3628.  @rwy0717
